dev_guide/dev_tutorials/binary_builds.adoc

```diff
@@ -0,0 +1,307 @@
+  to a local file named `Dockerfile`:
++
+----
+FROM wildfly:latest
+COPY ROOT.war /wildfly/standalone/deployments/ROOT.war
+CMD  $STI_SCRIPTS_PATH/run
+----
+
+. Create a new BuildConfig for that Dockerfile. 
++
+[NOTE]
+====
+This will automatically start a build that will initially fail because
+the ROOT.war artifact is not yet available. The pipeline below will pass that WAR to the build using a binary build.
+====
++
+----
+cat Dockerfile | oc new-build -D - --name mavenapp
+----
+
+. Create a BuildConfig with the Jenkins pipeline that will build a WAR and then use that WAR to build an image using the Dockerfile 
+  above. The same pattern can be used for other platforms where a binary artifact is built by a set of tools and
+  is then combined with a different runtime image for the final package.
+  Save the following code to `mavenapp-pipeline.yml`:
++
+----
+apiVersion: v1
+kind: BuildConfig
+metadata:
+  name: mavenapp-pipeline
+spec:
+  strategy:
+    jenkinsPipelineStrategy:
+      jenkinsfile: |-
+        pipeline {
+          agent { label "maven" }
+          stages {
+            stage("Clone Source") {
+              steps {
+                checkout([$class: 'GitSCM', 
+                            branches: [[name: '*/master']], 
+                            extensions: [
+                              [$class: 'RelativeTargetDirectory', relativeTargetDir: 'mavenapp']
+                            ], 
+                            userRemoteConfigs: [[url: 'https://github.com/openshift/openshift-jee-sample.git']]
+                        ])
+              }
+            }
+            stage("Build WAR") {
+              steps {
+                dir('mavenapp') {
+                  sh 'mvn clean package -Popenshift'
+                }
+              }
+            }
+            stage("Build Image") {
+              steps {
+                dir('mavenapp/target') {
+                  sh 'oc start-build mavenapp --from-dir . --follow'
+                }
+              }
+            }
+          }
+        }
+    type: JenkinsPipeline
+  triggers: []
+----
+
+. Create the pipeline build. If Jenkins is not deployed to your project, creating the buildconfig with the pipeline
+  will result in Jenkins getting deployed. It may take a couple of minutes before Jenkins is ready to build your pipeline.
+  You can check the status of the Jenkins rollout by invoking: `oc rollout status dc/jenkins`
++
+----
+oc create -f ./mavenapp-pipeline.yml
+----
+
+. Once Jenkins is ready, start the pipeline defined above
++
+----
+oc start-build mavenapp-pipeline
+----
+
+. When the pipeline has finished building, deploy the new application using new-app and expose its route:
++
+----
+oc new-app mavenapp
+oc expose svc/mavenapp
+----
+
+. Using your browser, navigate to the route for the application:
++
+----
+oc get route mavenapp
+----
```

Comment:
this ends a little abruptly.  no conclusion on what they've just done/seen?

Reply:
explanation added below